builder/builder.go

```diff
@@ -236,7 +241,24 @@ func (b *Builder) runStep(ctx context.Context, step *graph.Step) error {
 			step.Build = replacePositionalContext(step.Build, ".")
 		}
 		step.UpdateBuildStepWithDefaults()
-		args = b.getDockerRunArgs(volName, workingDirectory, step, step.Envs, "", "docker build "+step.Build)
+		useBuildCache := false
+
+		// buildx is only supported on Linux
+		if runtime.GOOS == linuxOS && step.UseBuildCacheForBuildStep() {
+			log.Println("runtime os is linux and need to use buildcache")
+			if buildStepWithBuildCache, err := step.FetchBuildWithBuildCache(); err != nil {
```

Comment:
is it easy to refactor this and put it in the Task processing/deserialization/initialization area instead? I never liked how the other one ended up here in the first place

Reply:
https://github.com/Azure/acr-builder/issues/472

yep i will. I could just move the methods/calling during task.initialize() but, i then i will have to fix a couple of tests bc we add "--isolation hyperv" stuff for windows but not on linux.